src/traces/isosurface/attributes.js

```diff
@@ -0,0 +1,184 @@
+        ].join(' ')
+    },
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Z coordinates of the vertices on Z axis.'
+        ].join(' ')
+    },
+    volume: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the 4th dimension of the vertices.'
+        ].join(' ')
+    },
+    isocap: {
+        valType: 'boolean',
+        role: 'info',
+        editType: 'calc',
+        dflt: false,
+        description: [
+            'Enable caps between minimum and maximum iso values.'
+        ].join(' ')
+    },
+    isovalue: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        dflt: [0],
+        description: [
+            'Sets iso surface boundaries.'
+        ].join(' ')
+    },
+    meshalgo: {
+        valType: 'string',
+        role: 'info',
+        dflt: 'MarchingCubes',
+        editType: 'calc',
+        description: [
+            'Sets the isosurface polygonizer algorithm:',
+            'including: `MarchingCubes` (i.e. default),',
+            '`MarchingTetrahedra` or `SurfaceNets`.'
+        ].join(' ')
+    },
+
+    text: {
+        valType: 'string',
+        role: 'info',
+        dflt: '',
+        arrayOk: true,
+        editType: 'calc',
+        description: [
+            'Sets the text elements associated with the vertices.',
+            'If trace `hoverinfo` contains a *text* flag and *hovertext* is not set,',
+            'these elements will be seen in the hover labels.'
+        ].join(' ')
+    },
+
+    // Color field
+    color: {
+        valType: 'color',
+        role: 'style',
+        editType: 'calc',
+        description: 'Sets the color of the whole isosurface'
+    },
+    vertexcolor: {
+        valType: 'data_array',
+        role: 'style',
+        editType: 'calc',
+        description: [
+            'Sets the color of each vertex',
+            'Overrides *color*.'
+        ].join(' ')
+    },
+    facecolor: {
+        valType: 'data_array',
+        role: 'style',
+        editType: 'calc',
+        description: [
+            'Sets the color of each face',
+            'Overrides *color* and *vertexcolor*.'
+        ].join(' ')
+    },
+    transforms: undefined
+},
+
+colorscaleAttrs('', {
+    colorAttr: '`intensity`',
+    showScaleDflt: true,
+    editTypeOverride: 'calc'
+}), {
+
+    colorbar: colorbarAttrs,
+
+    opacity: surfaceAtts.opacity,
+
+    // Flat shaded mode
+    flatshading: {
+        valType: 'boolean',
+        role: 'style',
+        dflt: false,
+        editType: 'calc',
+        description: [
+            'Determines whether or not normal smoothing is applied to the isosurfaces,',
+            'creating isosurfaces with an angular, low-poly look via flat reflections.'
+        ].join(' ')
+    },
+
+    contour: {
```

Comment:
`contour` doesn't appear to have been implemented (which is fine :smile: ) - let's :hocho: these attributes

Reply:
The contours seem to be working on hover namely on caps.